src/main/java/interface_adapter/search/SearchViewModel.java

```diff
@@ -2,13 +2,85 @@
 
 import interface_adapter.ViewModel;
 import interface_adapter.note.NoteState;
+import java.beans.PropertyChangeListener;
+import java.beans.PropertyChangeSupport;
 
 /**
- * The ViewModel for the NoteView.
+ * The ViewModel for the SearchView.
  */
 public class SearchViewModel extends ViewModel<NoteState> {
+
+    private String searchResults;
+    private String errorMessage;
+    private final PropertyChangeSupport propertyChangeSupport;
+
     public SearchViewModel() {
         super("note");
-        setState(new NoteState());
+        this.setState(new NoteState());
+        this.propertyChangeSupport = new PropertyChangeSupport(this);
+    }
+
+    /**
+     * Sets the search results and notifies observers of the change.
+     * @param searchResults the search results to display
+     */
+    public void setSearchResults(String searchResults) {
```

Comment:
Not sure how these will be used, since I don't often see code in ViewModels in the starter code, and they usually update the State of the ViewModels and firepropertychange to make Views react. But as long as it works for your use case, and it doesn't seem to violate Clean Architecture, it should be fine.